src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,57 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private K[] keyArray;
+    private V[] valueArray;
+    private int firstEmptyIndex;
+
+    public StorageImpl() {
+        firstEmptyIndex = 0;
+        keyArray = (K[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        valueArray = (V[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+    }
+
     @Override
     public void put(K key, V value) {
+        int containsIndex = contains(firstEmptyIndex, key);
```

Comment:
you can use your field directly
```suggestion
        int containsIndex = contains(key);
```